sktime/forecasting/base/_fh.py

```diff
@@ -220,9 +220,21 @@ def to_relative(self, cutoff=None):
 
         else:
             self._check_cutoff(cutoff)
-            values = self.to_pandas() - cutoff
+            index = self.to_pandas()
+
+            # Bug fix for DatetimeIndex delta calculation (see #534)
+            if isinstance(index, pd.DatetimeIndex):
+                try:
```

Comment:
What's the error that we need to catch here? Can we not check if either index or freq has a `freq` and raise an error otherwise?

Reply:
In the tests, there are some indices with only one value and `freq=None`, this is causing the error. Also we have to `try` first in order to convert e.g. the `MS` from `DatetimeIndex` into `M` of `PeriodIndex` automatically. If doing `y.index.to_period(freq="MS")` will result in the following error:
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-21-3c5e56bae74c> in <module>
----> 1 y.index.to_period(freq="MS")

c:\users\martin\miniconda3\envs\sktime37\lib\site-packages\pandas\core\indexes\extension.py in method(self, *args, **kwargs)
     76 
     77         def method(self, *args, **kwargs):
---> 78             result = attr(self._data, *args, **kwargs)
     79             if wrap:
     80                 if isinstance(result, type(self._data)):

c:\users\martin\miniconda3\envs\sktime37\lib\site-packages\pandas\core\arrays\datetimes.py in to_period(self, freq)
   1109             freq = res
   1110 
-> 1111         return PeriodArray._from_datetime64(self._data, freq, tz=self.tz)
   1112 
   1113     def to_perioddelta(self, freq):

c:\users\martin\miniconda3\envs\sktime37\lib\site-packages\pandas\core\arrays\period.py in _from_datetime64(cls, data, freq, tz)
    228         PeriodArray[freq]
    229         """
--> 230         data, freq = dt64arr_to_periodarr(data, freq, tz)
    231         return cls(data, freq=freq)
    232 

c:\users\martin\miniconda3\envs\sktime37\lib\site-packages\pandas\core\arrays\period.py in dt64arr_to_periodarr(data, freq, tz)
    957         data = data._values
    958 
--> 959     base = freq._period_dtype_code
    960     return c_dt64arr_to_periodarr(data.view("i8"), base, tz), freq
    961 

AttributeError: 'pandas._libs.tslibs.offsets.MonthBegin' object has no attribute '_period_dtype_code'
```

Reply:
> Can we not check if either index or freq has a freq and raise an error otherwise?

Then we have to rewrite some tests. And I think it is ok to accept also `freq=None` as e.g. an index like `pd.DatetimeIndex(['2000-02-10 00:00:00'])` is just resulting in `DatetimeIndex(['2000-02-10'], dtype='datetime64[ns]', freq=None)`